src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1591 @@
+-- Initialize harpstring variables to 0
+        A = "A"
+        B = "B"
+        C = "C"
+        D = "D"
+        E = "E"
+        F = "F"
+        G = "G"
+
+-----------------------------
+        local compare_notes = split(config.last_notes, ",")
+        for i, k in pairs(compare_notes) do
+            compare_notes[i] = trim(compare_notes[i])
+        end
+        --
+        local harp_tbl = split(harpnotes, ",")
+        local count = 0 
+
+        for i,k in pairs(harp_tbl) do
+            harp_tbl[i] = trim(harp_tbl[i])
+            if string.len(harp_tbl[i]) > 2 then
+                error = true
+                goto error1            
+            end
+            harp_tbl[i] = string.lower(harp_tbl[i])
+            local first = harp_tbl[i]:sub(1,1)
+            local second = harp_tbl[i]:sub(2,2)
+            if second == "f" then second = "b" end
+            if second == "s" then second = "#" end
+            if second == "n" then second = "" end
+            local first_upper = string.upper(first)
+            harp_tbl[i] = first_upper..second
+            if string.len(harp_tbl[i]) == 2 then
+                if string.sub(harp_tbl[i], -1) == "b" 
+                or string.sub(harp_tbl[i], -1) == "#" 
+                or string.sub(harp_tbl[i], -1) == "n" then
+                    -- then nothing!!!
+                else
+                    error = true
+                    goto error1     
+                end
+            end -- if length is 2...
+
+            ---- Assign to strings...
+            if harp_tbl[i]:sub(1,1) == "A" then 
+                A = harp_tbl[i]
+            elseif harp_tbl[i]:sub(1,1) == "B" then 
+                B = harp_tbl[i]
+            elseif harp_tbl[i]:sub(1,1) == "C" then
+                C = harp_tbl[i]
+            elseif harp_tbl[i]:sub(1,1) == "D" then
+                D = harp_tbl[i]
+            elseif harp_tbl[i]:sub(1,1) == "E" then
+                E = harp_tbl[i]
+            elseif harp_tbl[i]:sub(1,1) == "F" then
+                F = harp_tbl[i]
+            elseif harp_tbl[i]:sub(1,1) == "G" then
+                G = harp_tbl[i]
+            else
+                error = true
+                goto error1               
+            end -- End string assignments
+            count = i        
+        end -- for i,j
+--
+        harpstrings = {D, C, B, E, F, G, A}
+        pedals_update()
+
+        if count > 7 then
+            error = true
+            goto error1
+        end
+
+        if partial == true then
+            for i = 1, 7, 1 do
+                if harpstrings[i] == compare_notes[i] then
+                    new_pedals[i] = 0
+                else
+                    new_pedals[i] = harpstrings[i]
+                    changes = true
+                end
+            end
+        else
+            new_pedals = harpstrings
+        end
+
+        if (changes == false) and (partial == true) then
+            if direct == true then
```

Comment:
Again with the direct comparisons to boolean values. This is a habit I hope you will break.

```lua
if not changes or partial then
    if direct then
```
is cleaner

Reply:
The only times you might compare explicitly are

- to distinguish between `false` and `nil`. In this case I would compare to `nil`: `if x == nil then`
- to distinguish between `true` and some other non-nil non-boolean type. In this case, I would rewrite my code not to do this or put an assert on the type to guarantee it is a boolean type.

Reply:
I cleaned this all up, but had to leave the one in line 598